src/App.tsx

```diff
@@ -1,14 +1,51 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [loadingTodos, setLoadingTodos] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(loadedTodos => {
+        setLoadingTodos(false);
+        setTodos(loadedTodos);
+      });
+  }, []);
+
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState('all');
```

Comment:
move your `state` to the top, it will have better readability

Reply:
agree, have it all in one place